source/guides/maintenance/custom-app-container.rst

```diff
@@ -0,0 +1,461 @@
+      # image size, remove the overlapped files before copy.
+      RUN mkdir /os_core_install
+      -COPY --from=clearlinux/httpd:latest / /os_core_install/
+      +COPY --from=clearlinux/httpd:31090 / /os_core_install/
+      RUN cd / && \
+          find os_core_install | sed -e 's/os_core_install/install_root/' | xargs rm -d &> /dev/null || true
+
+      -FROM clearlinux/httpd:latest
+      +FROM clearlinux/httpd:31090
+
+#. Build Dockerfile.
+
+   .. code-block:: bash
+
+      docker build \
+      --no-cache \
+      --build-arg http_proxy=$http_proxy \
+      --build-arg https_proxy=$https_proxy \
+      -t clearlinux/cgit:31090 cgit/
+
+#. Verify the installed OS version by noting the :command:`VERSION_ID` value
+   in the :file:`/usr/lib/os-release` file in the container filesystem.
+
+   .. code-block:: bash
+      :linenos:
+      :emphasize-lines: 6
+
+      docker run clearlinux/cgit:31090 cat /usr/lib/os-release
+      NAME="Clear Linux OS"
+      VERSION=1
+      ID=clear-linux-os
+      ID_LIKE=clear-linux-os
+      VERSION_ID=31090
+      PRETTY_NAME="Clear Linux OS"
+      ANSI_COLOR="1;35"
+      HOME_URL="https://clearlinux.org"
+      SUPPORT_URL="https://clearlinux.org"
+      BUG_REPORT_URL="mailto:dev@lists.clearlinux.org"
+      PRIVACY_POLICY_URL=http://www.intel.com/privacy
+
+Example 4: Customize an application image at runtime
+====================================================
+
+This section describes how to modify a published |CL| container at runtime.
+In this example, we add Tensorflow\* into a :command:`clearlinux/python`
+container. This approach can help accelerate the feature development process.
+
+#. Launch the clearlinux/python container.
+
+   .. code-block:: bash
+
+      docker run -it --rm clearlinux/python
+      Python 3.7.3 (default, Jun 17 2019, 00:47:04)
+      [GCC 9.1.1 20190616 gcc-9-branch@272336] on linux
+      Type "help", "copyright", "credits" or "license" for more information.
+
+#. Try to import Tensorflow inside the container using the command:
+   :command:`import tensorflow as tf`. The example below shows the expected
+   error message because the Docker image does not yet include the Tensorflow module.
+
+   .. code-block:: bash
+
+      >>> import tensorflow as tf
+      Traceback (most recent call last):
+      File "<stdin>", line 1, in <module>
+      ModuleNotFoundError: No module named 'tensorflow'
+      >>>
+
+#. In another console, find the :command:`<Container_ID>` of
+   clearlinux/python launched. This example Container ID is d4ce9d526fa6.
+
+   .. code-block:: bash
+
+      docker ps
+
+#. The output shows:
+
+   .. code-block:: console
+
+     CONTAINER ID   IMAGE               COMMAND   CREATED             STATUS              PORTS          NAMES
+     d4ce9d526fa6   clearlinux/python   python3   About a minute ago  Up About a minute                  amazing_villani
+
+#. Connect to the running clearlinux/python container, and use

```

Comment:
@puneetse My instinct here is that there may be a step prior to this that's missing. 
When it states , "Connect to the running clearlinux/python container...." I tried to do so from within the Python repl, but I don't think that's correct. That may be why, when I try to do so, it fails.  When I run the command in Python repl, the error is a: Syntax Error
docker **exec**  -it <container_id> /usr/bin/bash

Error points to **exec** as source. Can you please clarify this concern or explain?

Reply:
Please see my email response from yesterday. You're right. Step #3 in that section should put you in a separate second console.

Reply:
Clarified. Revision adds new subheaders to clarify when to run new consoles.